internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -63,7 +63,7 @@ type Upgrader struct {
 func IsUpgradeable() bool {
 	// only upgradeable if running from Agent installer and running under the
 	// control of the system supervisor (or built specifically with upgrading enabled)
-	return release.Upgradeable() || (paths.RunningInstalled() && info.RunningUnderSupervisor())
+	return release.Upgradeable() || (paths.InstallMarkerExists() && info.RunningUnderSupervisor())
```

Comment:
Same as https://github.com/elastic/elastic-agent/pull/4172/files#r1480469016. Let's add a comment that paths.InstallMarkerExists() exists to mark RPM and DEB packages as not upgradable from the UI.

Also if we could stop duplicating this logic that would be nice.